pyproject.toml

```diff
@@ -26,6 +26,7 @@ quimb = { version = "^1.6.0", extras = ["tensor"] }
 cupy-cuda11x = { version = "^11.6.0", optional = true }
 cuquantum-python-cu11 = { version = "^23.3.0", optional = true }
 mpi4py = { version = "^3.1.5", optional = true }
+qibojit = { git = "https://github.com/qiboteam/qibojit.git" }
```

Comment:
Why is Qibotn now depending on Qibojit?

Reply:
Qibotn had qibojit dependency earlier in the tests. So far we were installing qibojit seperately. Since, the previous version of workflow was skipping tests it was not an issue in github actions. But, now in the absence of qibojit tests will fail.

Reply:
But you should not run tests on Qibojit, since you're explicitly testing another backend (i.e. Qibotn).

Could you point me to the logs where tests are failing?

Reply:
I guess the question here is "Does Qibotn need to depend on Qibojit?". My quick answer is "No, it doesn't."

So I removed `qibojit` package, ran `pytest`, and encountered an error below.
```
        # Test qibo
>       qibo.set_backend(backend=config.qibo.backend, platform=config.qibo.platform)

tests/test_quimb_backend.py:45:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../miniforge3/envs/py311/lib/python3.11/site-packages/qibo/backends/__init__.py:150: in set_backend
    GlobalBackend.set_backend(backend, **kwargs)
../../miniforge3/envs/py311/lib/python3.11/site-packages/qibo/backends/__init__.py:108: in set_backend
    cls._instance = construct_backend(backend, **kwargs)
../../miniforge3/envs/py311/lib/python3.11/site-packages/qibo/backends/__init__.py:230: in construct_backend
    raise_error(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

exception = <class 'ValueError'>
message = "The 'qibojit' backends' provider is not available. Check that a Python package named 'qibojit' is installed, and it is exposing valid Qibo backends."
```

I guess it's due to `qibojit` being the first and default choice during backend construction. If my guess is correct, maybe we specify the backend to be `numpy` for test cases of `quimb` instead of using the default backend?

Reply:
> I guess it's due to `qibojit` being the first and default choice during backend construction. If my guess is correct, maybe we specify the backend to be `numpy` for test cases of `quimb` instead of using the default backend?

This is a known problem, already solved recently https://github.com/qiboteam/qibo/pull/1430

The point is that you should not encounter that problem, since you should set Qibotn as the backend, and not relying on the default one.

Reply:
Okay, will modify the tests to remove the dependency